Document workflow execution permissions (#7137)
elastic/docs-content: explore-analyze/workflows/concepts.md

Reviewer: I'd suggest to make this its own page rather than a section on `concepts.md` since this one is more of a small TOC page:

- New page: `explore-analyze/workflows/execution-permissions.md`
- Add it in `explore-analyze/toc.yml` as a child of `workflows/concepts.md`, alongside triggers/steps/templating/reference.

Reviewer: Suggest adding reciprocal links to make execution permissions discoverable from the pages that depend on it. Right now only `scheduled-triggers.md` links to it. (link suggestions here assume we're making this section a page as per my comment above)

`triggers/alert-triggers.md` — after the intro paragraph:
```markdown
Alert-triggered workflows run with the permissions of the user who created or last modified the rule. Refer to [](/explore-analyze/workflows/execution-permissions.md) to learn how workflow execution permissions work.
```

`triggers/event-driven-triggers.md` — as a bullet in the `## Related` list:
```markdown
- [Execution permissions](/explore-analyze/workflows/execution-permissions.md): Which user context event-driven workflows run with.
```

`reference/context-variables.md` — under the `execution.*` table:
```markdown
For how `execution.executedBy` is determined for each trigger type, refer to [Execution permissions](/explore-analyze/workflows/execution-permissions.md).
```

And from the execution-permissions content, link back to `context-variables.md#workflows-ctx-execution` as the canonical reference for `execution.executedBy` / `execution.triggeredBy` (rather than restating their definitions).

- [Steps](/explore-analyze/workflows/steps.md): Action, flow control, AI, data, and composition steps.
- [Templating engine](/explore-analyze/workflows/templating.md): Liquid templating syntax and custom filters.
- [Reference](/explore-analyze/workflows/reference.md): Quick-reference pages for the workflow YAML surface. Cheat sheet, A-Z step type index, context variables, and the Liquid filter catalog.

## Execution permissions [workflows-execution-permissions]

> Workflows execute with a {{kib}} user context. That context determines which {{kib}} and {{es}} resources the workflow can access during the run. Connector steps use credentials stored in the connector to authenticate to the external system.

Reviewer: ```suggestion
Workflows execute with a {{kib}} user context. That context determines which {{kib}} and {{es}} resources the workflow can access during the run. [Connector-based action steps](/explore-analyze/workflows/steps/external-systems-apps.md#connector-based-actions) authenticate to the external system with the credentials stored in the connector.
```
Looks like this is the terminology in use in the docs

> The execution identity depends on how the workflow is triggered:

Reviewer: **Question: what happens when the credential owner is deactivated or loses privileges?**

For scheduled and alert runs, the execution uses the stored credentials of a specific user. It's probably worth explicitly documenting the behavior when that user is later deactivated, deleted, or loses a privilege a step needs: do future runs fail outright, or do individual steps fail with authorization errors?

Author: API keys live even if the user who created them changes/gets deactivated

> * Manual runs from the {{kib}} UI or API use the permissions of the logged-in user who starts the run.

Reviewer: ```suggestion
* Manual runs from the {{kib}} UI or API use the permissions of the authenticated user who starts the run.
```
Nit: Works slightly better as a verb for both UI and API (maybe a bit subjective 😄)

> * Scheduled workflows use the permissions of the user who last modified the workflow. When an enabled scheduled workflow is edited, {{kib}} refreshes the stored execution credentials for future runs.

Reviewer: ```suggestion
* Scheduled workflows use the permissions of the user who last saved the workflow. When an enabled scheduled workflow is edited, {{kib}} refreshes the stored execution credentials for future runs.
```
same as my suggestion below to avoid the "either modified or created if not modified" long blurb

> * Alert and detection rule triggers use the permissions of the user who last modified the rule. If the rule has not been modified, they use the permissions of the user who created the rule.

Reviewer: ```suggestion
* Alert and detection rule triggers use the permissions of the user who last saved the rule.
```
Would that work to make it shorter?

> * Event-based triggers use the permissions of the user whose action produced the event. For example, a `cases.addComment` event runs the workflow with the permissions of the user who added the comment.

Reviewer: ```suggestion
* Event-based triggers use the permissions of the user whose action produced the event. For example, a `cases.commentsAdded` event runs the workflow with the permissions of the user who added the comment.
```
IIUC that's the documented event type whereas cases.addComment is a workflow step type

Is this worth adding since in the event trigger docs we distinguish case triggers and workflows.failed? -> ``For `workflows.failed`, the handler runs with the permissions the failed workflow ran with.``

When a workflow run starts, {{kib}} records execution metadata:

* `execution.executedBy`: the user whose permissions are used for the workflow run.

> * `execution.triggeredBy`: the trigger that started the workflow, such as `manual` or `scheduled`.

Reviewer: ```suggestion
When a workflow run starts, {{kib}} records the execution identity so you can audit which permissions a run used:

* `execution.executedBy`: the user whose permissions the run uses.
* `execution.triggeredBy`: the trigger that started the run, such as `manual` or `scheduled`.

Refer to [](/explore-analyze/workflows/reference/context-variables.md#workflows-ctx-execution) for the full execution metadata reference.
```
suggesting to add a link to the source of truth for reference
